ext/tokenizer/tokenizer.c

```diff
@@ -408,7 +410,7 @@ static zend_bool tokenize(zval *return_value, zend_string *source, zend_class_en
 				&& --need_tokens == 0
 			) {
 				/* fetch the rest into a T_INLINE_HTML */
-				if (zendcursor != zendlimit) {
```

Comment:
This breaks if it receives a `T_ERROR` token at end of file (which was not possible before). I'm not sure why, but in that case, `zendcursor` is actually *more* than `zendlimit`. Changing the test to `zendcursor < zendlimit` makes it a bit more robust.

The added guard clause above actually takes care of this problem, but I figure changing the test to `zendcursor < zendlimit` is still a good idea.